Gems/Multiplayer/Code/Source/Debug/MultiplayerConnectionViewportMessageSystemComponent.cpp

```diff
@@ -0,0 +1,146 @@
+    {
+        AZ::RPI::ViewportContextPtr viewport = AZ::RPI::ViewportContextRequests::Get()->GetDefaultViewportContext();
+        if (!viewport)
+        {
+            return;
+        }
+
+        // Display the custom center viewport text
+        if (!m_centerViewportDebugText.empty())
+        {            
+            AzFramework::WindowSize viewportSize = viewport->GetViewportSize();
+            const float dpiScalingFactor = viewport->GetDpiScalingFactor();
+            const float center_screenposition_x = 0.5f*viewportSize.m_width;
+            const float center_screenposition_y = 0.5f*viewportSize.m_height;
+            const float screenposition_title_y = center_screenposition_y-9;
+            const float screenposition_debugtext_y = screenposition_title_y+18*dpiScalingFactor;
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::Yellow);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_title_y / dpiScalingFactor, editorsv_connectionMessageFontSize, "Multiplayer Editor", true);
+
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::SetColor, AZ::Colors::White);
+            AzFramework::DebugDisplayRequestBus::Broadcast(&AzFramework::DebugDisplayRequestBus::Events::Draw2dTextLabel, 
+                center_screenposition_x / dpiScalingFactor, screenposition_debugtext_y / dpiScalingFactor, editorsv_connectionMessageFontSize, m_centerViewportDebugText.c_str(), true);
+        }
+
+        // Build the connection status string (just show client connected or disconnected status for now)
+        const auto multiplayerSystemComponent = AZ::Interface<IMultiplayer>::Get();
+        MultiplayerAgentType agentType = multiplayerSystemComponent->GetAgentType();        
+        if (agentType != MultiplayerAgentType::Client)
+        {
+            return;
+        }
+
+        // Display the connection status to the top-left viewport
+        if (AzNetworking::INetworkInterface* networkInterface = AZ::Interface<AzNetworking::INetworking>::Get()->RetrieveNetworkInterface(AZ::Name(MpNetworkInterfaceName)))
+        {
+            AzNetworking::IConnectionSet& connectionSet = networkInterface->GetConnectionSet();
+            if (connectionSet.GetConnectionCount() > 0)
+            {
+                // Display the connection status (calling VisitConnections(), but there's only 1 since we're a client)
+                auto displayConnectionStatus = [](AzNetworking::IConnection& connection)
+                {
+                    AZStd::string connectionStatusText = AZStd::string::format("Operating as multiplayer Client: %s", GetEnumString(connection.GetConnectionState()));
```

Comment:
```suggestion
                    AZStd::string connectionStatusText = AZStd::string::format("Operating as multiplayer Client: %s", ToString(connection.GetConnectionState()));
```

ConnectionState is an AZ_ENUM_CLASS so you can use its constexpr ToString.

Reply:
Also, if this is strictly read only and limited to this scope (including bus usage) you could use AZStd::string_view over AZStd::string.

Reply:
fixed!

Reply:
None of it will actually end up compile-time though. Const-expr won't offer any meaningful benefit in these cases. I would just do whatever is simplest and least likely to cause errors moving forward. Potentially non-null terminated .data() being fed into format strings is a pretty big no-no..

Reply:
I was actually thinking there might be an extra copy here but it could be a misread on my part.